### PR TITLE
don't dereference prefault_dbp if send_prefault_udp is off

### PR DESCRIPTION
**Pull request by adizaimi**

### CODE REVIEW

**Comment by riverszhang89** [comment]
Skip build for now. I am testing smoke test.

File: berkdb/mp/mp_fget.c [hunk 1]
```diff
@@ -1174,52 +1174,47 @@ __memp_fget(dbmfp, pgnoaddr, flags, addrp)
...
-						   to it. Add the page. */
-						__memp_add_sparse_page(dbmfp->dbenv,
-								prefault_dbp->log_filename->id,
-								prefault_dbp->fileid,
-								*pgnoaddr, sparseness);
-					}
-				}
-			}
-		}
+    ret = __memp_fget_internal(dbmfp, pgnoaddr, flags, addrp, &did_io);
```
Comment (akshatsikarwar) [inline berkdb/mp/mp_fget.c]: Use tabstop=8 noet sw=8 for berkdb code

**Comment by ldennis** [comment]
LGTM

**Comment by adizaimi** [comment]
online_compaction is failing so i wonder if this change is affecting it.

**Comment by riverszhang89** [comment]
Does online_compaction pass if you run it manually?

**Comment by adizaimi** [comment]
Does not pass running manually. Debugging via printf, seems that function __memp_fget() is called with pgcomp_enabled=1 and prefault_dbp=NULL.

**Comment by riverszhang89** [comment]
Looks like we can't simply set prefault_dbp to NUL before exiting memp_fget(). The dbp may get pages multiple times and only the 1st time we may be able to trigger page prefault or compaction.

**Comment by riverszhang89** [comment]
Looks great! Thanks Adi!

**Pull request merged**